### DISCUSSION TITLE
Problema nello scalare un immagine ad alta definizione in un canvas di dimensioni piccole

### DISCUSSION THREAD

**Discussion by NicolaCocozza** [fabricjs/fabric.js#9714] [Q&A]
Salve a tutti, nel mio progetto ho un canvas di dimensioni 140px(lunghezza) per 140px(altezza), quando carico un immagine di dimensione 5514px(lunghezza) per 2500px(altezza) con 300 di dpi, utilizzo la libreria fabric.js per eseguire lo scaling dell'immagine. L'immagine generata ha giustamente le dimensioni corrette ma la definizione non è quella corretta. Vorrei sapere se sto sbagliando qualcosa o devo impostare qualche attributo come objectCaching e enableRetinaScaling, oppure è un errore irrisolvibile? La versione che ho è la 4.4.0, vi ringrazio già in anticipo

**Comment by asturur** [comment]
Hello Nicola, while i m Italian and speak Italian, the official language of the project is English.

When you say `L'immagine generata ha giustamente le dimensioni corrette ma la definizione non è quella corretta` and so the image generated doesn't look good, could you please show what you mean with a screenshot?

If the resize is bad you may want to use one of the resize filter available, otherwise you may want to look in some properties of drawImage that helps with quality.
It also depends if you are running on node or browser.

If you can't write in english is ok to use an automatic translator.
